### DISCUSSION TITLE
Why different selection "modes" and where do the behavior differences come from?

### DISCUSSION THREAD

**Discussion by Stebalien** [meow-edit/meow#388] [Q&A]
This is a bit of a composite question: Why different selection modes and why to they seem to behave slightly differently:

Word/line behave as separate selection "modes", switching between them and/or hitting `meow-$dir` exits the selection. So far we're consistent, and this seems quite useful (e.g., `meow-line` followed by `meow-back-word` to select the last word on a line). But I'm wondering if there's a way to keep the current selection?

`meow-$dir-expand` _keeps_ the selection, but switches to char selection mode. This is quite useful, I just wish I could switch _back_ to word/line selection. E.g.,

- Run `meow-line` to select the current line, and re-enter line selection mode.
- Run `meow-line` then `meow-back-word` to select the whole line, minus the last word.

When in _char_ selection mode, `meow-$dir` _keeps_ char selection. This one has me pretty confused, actually.

`meow-$dir-expand` (and even `meow-line`) moves the edge of the selection, while `meow-next/back-word` _expand_ the selection by a word. This one also has me a bit confused:

- Even if `meow-next/back-word` behaved the same way as `meow-$dir-expand`, I could use `meow-reverse` to expand in either direction.
- I'd love to be able to:
   - Select an entire symbol.
   - Switch to the front of the symbol.
   - Unselect the first word.

----

I'm mostly looking for context and/or use-cases/motivations that I might be missing.

**Comment by DogLooksGood** [comment]
Hey!

Actually, based on the frequencies of usages, Meow divides movement commands into 3 groups. 
1. The most used are char-based movements, so each direction, expand and non-expand all have their own commands.t e.g. `meow-left`, `meow-left-expand`.
2. The second group is word/symbol movements, so backward and forward are separated. And there's one more command for selecting the whole thing. e.g. `meow-next-word`, `meow-mark-word`.
3. Rest are not frequently used, so for each type, there's only one command. e.g. `meow-line`.

And the rules are here:
https://github.com/meow-edit/meow/blob/master/EXPLANATION.org#selection

And you don't really need a way to keep current selection. Because to select something, you only need the start position where your cursor is, and the target position where you can describe with Meow's command. e.g. "To the 7th word", "To the 3rd line". For arbitrary selection, just use `meow-$dir-expand`, then you can use `meow-$dir` to create arbitrary selection, based on the rule 2&3. Or just use the Emacs built-in commands(`M-f`, `C-f`, etc), they are always available.
